src/oaklib/implementations/semsimian/semsimian_implementation.py

```diff
@@ -136,6 +157,8 @@ def all_by_all_pairwise_similarity(
         """
         objects = list(objects)
         logging.info(f"Calculating all-by-all pairwise similarity for {len(objects)} objects")
+        if not hasattr(self, "semsimian") or getattr(self, "semsimian", None) is None:
```

Comment:
this can be simplified to `if getattr(self, "semsimian", None) is None`

Reply:
or even simpler: `if self.semsimian is None`

Reply:
`if` condition not needed anymore with caching in place.

Reply:
Resolved